### PR TITLE
[auto] using rector to upgrade code base to php 8.1 (php-min version)

### PR DESCRIPTION
**Pull request by nusphere**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since PHP 8.1 is the min PHP version, this is the time to upgrade the code base to PHP 8.1 code. I let rector do the drill...

### CODE REVIEW

**Comment by samsonasik** [comment]
You should can run:

```
vendor/bin/phpcbf
```

to fix cs.

**Comment by nusphere** [comment]
Need some time to figure out, how to handle the psalm errors. 

Next time window, for fixing them, is tomorrow.

- [ ] handle open psalm issues

**Comment by nusphere** [comment]
@samsonasik 
Thank you for your advice, I keep going. 😉

File: test/_files/ModuleWithEvents/Module.php [hunk 1]
```diff
@@ -14,7 +14,7 @@ public function onBootstrap(MvcEvent $e)
     {
         $application = $e->getApplication();
         $events      = $application->getEventManager();
-        $events->attach(MvcEvent::EVENT_ROUTE, [$this, 'onRoute'], -1000);
+        $events->attach(MvcEvent::EVENT_ROUTE, $this->onRoute(...), -1000);
```
Comment (Xerkus) [inline test/_files/ModuleWithEvents/Module.php:17]: This change is invalid

Comment (Xerkus) [reply test/_files/ModuleWithEvents/Module.php:17]: Hm. I just now realized what this does. I totally forgot it was a thing. I am not convinced we should actually use it. It is hard to read.

**Pull request closed without merging**